lib/active_snapshot/models/concerns/snapshots_concern.rb

```diff
@@ -9,32 +9,16 @@ module SnapshotsConcern
     end
 
     def create_snapshot!(identifier: nil, user: nil, metadata: nil)
-      snapshot = snapshots.create!({
-        identifier: identifier,
-        user_id: (user.id if user),
-        user_type: (user.class.name if user),
-        metadata: (metadata || {}),
-      })
-
-      new_entries = []
+      snapshot = Snapshot.build_snapshot(self, identifier: identifier, user: user, metadata: metadata)
+      new_entries = snapshot.snapshot_items.map(&:attributes)
+      snapshot.snapshot_items.reset # clear the association cache otherwise snapshot.valid? returns false
```

Comment:
Is there a reason that we are changing the order in which the `reset`  method is called?

Reply:
as I said before, I need to move this up so that I can save the snapshot with validations